### PR TITLE
Tiogia TPL Update / Build Fixes

### PR DESCRIPTION
**Pull request by wrtobin**
- [x] Update to more recent tioga TPLs (https://github.com/GEOS-DEV/LvArray/pull/288)
- [x] Fix tioga fmt link-time error
- [x] Add tioga to nightly gitlab CI so we have at least a modicum of HIP/ROCM automated CI coverage (https://github.com/GEOS-DEV/nightlyTests/pull/7)

### CODE REVIEW

File: host-configs/LLNL/quartz-clang@14.cmake [hunk 1]
```diff
@@ -1,5 +1,7 @@
 include(${CMAKE_CURRENT_LIST_DIR}/../../src/coreComponents/LvArray/host-configs/LLNL/quartz-clang@14.cmake)
 
+set(CMAKE_CXX_FLAGS "-ftime-trace" CACHE STRING "" )
```
Comment (wrtobin) [inline host-configs/LLNL/quartz-clang@14.cmake:3]: Whoops gotta remove this.

File: src/coreComponents/dataRepository/xmlWrapper.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ void stringToInputVariable( Tensor< T, SIZE > & target, string const & inputValu
   std::istringstream ss( inputValue );
   auto const errorMsg = [&]( auto const & msg )
   {
-    return GEOS_FMT( "{} for Tensor<{}> at position {} in input: {}", msg, SIZE, ss.tellg(), inputValue );
+    return GEOS_FMT( "{} for Tensor<{}> at position {} in input: {}", msg, SIZE, static_cast< int >( ss.tellg() ), inputValue );
```
Comment (wrtobin) [inline src/coreComponents/dataRepository/xmlWrapper.cpp:38]: Something about the `tellg` return time isn't handled appropriately by CCE on tioga using the more modern fmt libraray, but by the standard it is convertible to integer types, so this resolves the compile failure.

File: src/docs/sphinx/buildGuide/BuildProcess.rst [hunk 3]
```diff
@@ -68,10 +68,12 @@ Option                          Default   Explanation
 ``ENABLE_OPENMP``               ``OFF``   Build with OpenMP (also applies to TPLs)
 ``ENABLE_CUDA``                 ``OFF``   Build with CUDA (also applies to TPLs)
 ``ENABLE_CUDA_NVTOOLSEXT``      ``OFF``   Enable CUDA NVTX user instrumentation (via GEOS_MARK_SCOPE or GEOS_MARK_FUNCTION macros)
+``ENABLE_HIP``                  ``OFF``   Build with HIP/ROCM (also applies to TPLs)
 ``ENABLE_DOCS``                 ``ON``    Build documentation (Sphinx and Doxygen)
 ``ENABLE_WARNINGS_AS_ERRORS``   ``ON``    Treat all warnings as errors
 ``ENABLE_PVTPackage``           ``ON``    Enable PVTPackage library (required for compositional flow runs)
 ``ENABLE_TOTALVIEW_OUTPUT``     ``OFF``   Enables TotalView debugger custom view of GEOS data structures
+``GEOS_ENABLE_TESTS``           ``ON``    Enables unit testing targets
```
Comment (wrtobin) [inline src/docs/sphinx/buildGuide/BuildProcess.rst:76]: This was supposed to be in the unit test PR, but managed to be forgotten.

**Comment by bmhan12** [comment]
tioga needs to be added here to "paths" so gitlab will save the created folder of artifacts for nightlyTests. 
https://github.com/GEOS-DEV/GEOS/blob/461dadf0beb2e115b4fec05bd5b6dbd4f4b68fd3/.gitlab-ci.yml#L91-L98

Additionally, on the nightlyTests repository side of things, some scripting files need to modified so the tioga artifacts will be scraped and front-facing README updated:

Text file keeping track of when different phases were last passing. Entries need to be added for tioga, expected by the `update.sh` script.
https://github.com/GEOS-DEV/nightlyTests/blob/master/last_passed.txt

Add tioga to list of systems.
https://github.com/GEOS-DEV/nightlyTests/blob/3e8aadd1e6f9c522e5e1c67a6f628f45c4be2e40/update.sh#L19

**Comment by wrtobin** [comment]
@bmhan12 Just opened a PR on nightlyTests with the noted changes: https://github.com/GEOS-DEV/nightlyTests/pull/7

**Comment by wrtobin** [comment]
Would it break anything in the gitlab CI if altered the sha1 hashes in last_passed to `0000000` instead of just some copied hashes ( I assume we don't have any verification of those hashes, but you'd know better than I do ).

**Comment by bmhan12** [comment]
> Would it break anything in the gitlab CI if altered the sha1 hashes in last_passed to 0000000 instead of just some copied hashes ( I assume we don't have any verification of those hashes, but you'd know better than I do ).

Nope, no verification. In the past, I think I've put TBD or similar placeholder.

**Comment by wrtobin** [comment]
@CusiniM @rrsettgast 

I was hoping to merge this just now but had a 3 hour timeout because of course. If I'm unable to merge it later on tonight and someone else gets to it first, once it goes in we can merge the nightlyTests PR linked above (it shouldn't go in first since it isn't a submodule and depends on this PR being merged to function correctly).

**Pull request merged**